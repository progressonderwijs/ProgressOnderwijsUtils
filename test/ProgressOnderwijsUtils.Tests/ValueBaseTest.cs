using System;
using JetBrains.Annotations;
using Xunit;

namespace ProgressOnderwijsUtils.Tests
{
    [UsedImplicitly(ImplicitUseTargetFlags.Members)] // to test valuebase.ToString
    public sealed class ExampleValue : ValueBase<ExampleValue>
    {
        public int MyInt;
        public string MyString { get; set; }
        public DateTime SomeValueType { get; set; }
        public ExampleValue Nested;
        public double? NullableField;
        public ConsoleKey? AnEnum;
    }

    public sealed class ValueBaseTest
    {
        [Fact]
        public void ToString_ReturnsCleanLookingOutput()
        {
            ApprovalTest.Verify(
                new ExampleValue {
                    NullableField = null,
                    AnEnum = ConsoleKey.BrowserBack,
                    MyString = "Hello World!",
                    Nested = new ExampleValue { AnEnum = ConsoleKey.BrowserRefresh },
                    SomeValueType = default(DateTime),
                    MyInt = 42,
                }.ToString());
        }
    }
}
