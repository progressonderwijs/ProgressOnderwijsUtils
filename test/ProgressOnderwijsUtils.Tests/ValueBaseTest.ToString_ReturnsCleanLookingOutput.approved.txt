new ExampleValue {
    MyInt = 42,
    Nested = new ExampleValue {
        MyInt = 0,
        Nested = null,
        NullableField = null,
        AnEnum = ConsoleKey.BrowserRefresh,
        MyString = null,
        SomeValueType = default(DateTime),
    },
    NullableField = null,
    AnEnum = ConsoleKey.BrowserBack,
    MyString = "Hello World!",
    SomeValueType = default(DateTime),
}